.github/workflows/check-for-wheel-build.yml

```diff
@@ -0,0 +1,42 @@
+name: Check if Wheels Build Job should run
+on:
+  workflow_call:
+    outputs:
+      build-wheels:
+        description: Indicate whether wheels should be built for the current workflow run
+        value: ${{ jobs.check_if_required.outputs.build-wheels }}
+
+
+jobs:
+  check_if_required:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Build is triggered from Pull Request
+        id: is_pr
+        run: echo "is_pr=${{ github.event_name == 'pull_request' }}" >> $GITHUB_OUTPUT
+
+      - name: Pull Request Build needs Wheel Builds to pass
+        id: needs_wheel_builds
+        if: steps.is_pr.outputs.is_pr == 'true'
+        run: echo "needs_wheel_builds=${{ contains(github.event.pull_request.labels.*.name, 'requires-wheel-builds') }}" >>$GITHUB_OUTPUT
+
+      - name: Build Wheels for Pull Request
+        id: build_wheels
+        if: steps.is_pr.outputs.is_pr == 'true'
+        run: echo "build_wheels=${{ contains(github.event.pull_request.labels.*.name, 'ci:build-wheels') }}" >> $GITHUB_OUTPUT
+
+      # If a pr has the `requires-wheel-builds` label, that means that the Workflows which build wheels need to successfully run against it
+      # However, the PR does not have the `ci:build-wheels` label, meaning it is not ready for the wheel workflows to run against it yet.
+      # In that condition, this step will fail, causing the entire workflow to fail.
+      # And since this job is required to pass on all jobs, it will cause the merging of the pull request to be blocked.
+      - name: Fail for Pull Request that needs wheels built but does not have build-wheels label
+        if: steps.is_pr.outputs.is_pr == 'true' && steps.needs_wheel_builds.outputs.needs_wheel_builds == 'true' && steps.build_wheels.outputs.build_wheels == 'false'
+        run: echo ::error title=Failing for Pull Request that needs wheel builds with build-wheels label missing::This pull request requires the wheels to build successfully against it. Add the build-wheels label. && exit 1
```

Comment:
a minor suggetion: you may want to consider using `ci:build-wheels` instead of `build-wheels` when referring to the lable.

Reply:
I'll just close this to merge the PR, seems fairly minor.

Reply:
I will fix this in a subsequent PR :+1: